### PR TITLE
chore(mojaloop/#3455): nodejs upgrade

### PR DESCRIPTION
**Pull request by oderayi**
chore(mojaloop/#3452): NodeJS version upgrade - https://github.com/mojaloop/project/issues/3452
- Upgraded NodeJS from v16 to v18 LTS
- Updated CI config to use NodeJS latest LTS version
- Updated CI config to display runtime versions for each job
- Upgraded NPM dependencies
- Fixed CI bugs caused by the NodeJS upgrade
- Upgraded CircleCI orbs versions
- Updated CI pipeline use NVM to select node version from` .nvmrc` file
- Replaced [gRPC](https://www.npmjs.com/package/grpc) with [gRPC-js](https://www.npmjs.com/package/@grpc/grpc-js) 
- optimized logging calls

### CODE REVIEW

**Review by kleyow — APPROVED** [review]

**Review by kleyow — DISMISSED** [review]

File: src/transport/EventLoggingServiceClient.ts [hunk 1]
```diff
@@ -54,9 +54,9 @@ class EventLoggingServiceClient {
 
         const wireEventCopy: any = JSON.parse(JSON.stringify(wireEvent));
         wireEventCopy.content.value = `Buffer(${wireEventCopy.content.value.data.length})`
-        Logger.debug(`EventLoggingServiceClient.log sending wireEvent: ${JSON.stringify(wireEventCopy, null, 2)}`);
+        Logger.isDebugEnabled && Logger.debug(`EventLoggingServiceClient.log sending wireEvent: ${JSON.stringify(wireEventCopy, null, 2)}`);
```
Comment (mdebarros) [inline src/transport/EventLoggingServiceClient.ts:57]: Can we optimise the above wirecopy?

I.e. as far as I can see the wirecopy is only being used for the debug statements?

If so let's not do a copy (assuming it's even necessary?) unless we are going to log the debug statements.

Comment (oderayi) [reply src/transport/EventLoggingServiceClient.ts:57]: Then I would wrap all of that within an `if (Logger.isDebugEnabled) { ... }`

File: src/transport/EventLoggingServiceClient.ts [hunk 2]
```diff
@@ -52,9 +52,10 @@ class EventLoggingServiceClient {
       try {
         wireEvent.content = toAny(event.content, event.type);
 
-        const wireEventCopy: any = JSON.parse(JSON.stringify(wireEvent));
-        wireEventCopy.content.value = `Buffer(${wireEventCopy.content.value.data.length})`
-        Logger.isDebugEnabled && Logger.debug(`EventLoggingServiceClient.log sending wireEvent: ${JSON.stringify(wireEventCopy, null, 2)}`);
+        if (Logger.isDebugEnabled) {
+          const wireEventCopy = {...wireEvent, content: {...wireEvent.content, value: `Buffer(${wireEvent.content.value.data.length})`}} 
+          Logger.isDebugEnabled && Logger.debug(`EventLoggingServiceClient.log sending wireEvent: ${JSON.stringify(wireEventCopy, null, 2)}`);
```
Comment (mdebarros) [inline src/transport/EventLoggingServiceClient.ts:57]:
```suggestion
          Logger.debug(`EventLoggingServiceClient.log sending wireEvent: ${JSON.stringify(wireEventCopy, null, 2)}`);
```

**Review by mdebarros — APPROVED** [review]
+1

**Pull request merged**